In [11]:
    library(alr3)
    library(model) 

## Ejercicio F:

### Ejercicio F.1

In [3]:
X_1 <- c(17, 19, 19, 20, 20, 20, 20, 21, 21, 21 ,21, 25, 25, 27, 28, 30, 30)
X_2 <- c(42, 45, 45, 29, 29, 29, 29, 93, 93, 93, 93, 34, 34, 98, 9, 73, 73)
Y <- c(90, 71, 76, 63, 63, 80, 80, 80, 64, 82, 66, 75, 82, 99, 73, 67, 74)
data_F <- data.frame(X_1, X_2, Y)
data_F

X_1 X_2 Y 
1  17  42  90
2  19  45  71
3  19  45  76
4  20  29  63
5  20  29  63
6  20  29  80
7  20  29  80
8  21  93  80
9  21  93  64
10 21  93  82
11 21  93  66
12 25  34  75
13 25  34  82
14 27  98  99
15 28   9  73
16 30  73  67
17 30  73  74

In [4]:
regresion_F.1 <- lm(Y ~ X_1+X_2, data_F)
summary(regresion_F.1)


Call:
lm(formula = Y ~ X_1 + X_2, data = data_F)

Residuals:
     Min       1Q   Median       3Q      Max 
-13.3764  -9.6597   0.3827   5.7693  21.2085 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  72.2462    14.9067   4.847 0.000259 ***
X_1           0.0286     0.6455   0.044 0.965279    
X_2           0.0487     0.0876   0.556 0.586984    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 10.38 on 14 degrees of freedom
Multiple R-squared:  0.02267,	Adjusted R-squared:  -0.1169 
F-statistic: 0.1624 on 2 and 14 DF,  p-value: 0.8517


Este modelo ta re weno xd.

In [5]:
coef(`regresion_F.1`)

## Ejercicio K:

In [8]:
F.aov <- aov(Y ~ X_1+X_2,data_F)
summary(F.aov)

            Df Sum Sq Mean Sq F value Pr(>F)
X_1          1    1.7    1.69   0.016  0.902
X_2          1   33.3   33.32   0.309  0.587
Residuals   14 1509.1  107.79               

El modelo queda Y= 72.2462+0.0286*X_1+0.0487*X_2

In [28]:
I=diag(1,nrow=3)
C<-matrix(c(5,0,0,0,4,2,0,2,2), nrow = 3)
solve(C,I)

[,1] [,2] [,3]
[1,] 0.2   0.0  0.0
[2,] 0.0   0.5 -0.5
[3,] 0.0  -0.5  1.0

In [29]:
J=diag(1,nrow=3)
C2<- matrix(c(4.25,0,0,0,4,2,0,2,2),nrow=3)
solve(C2,J)

[,1]      [,2] [,3]
[1,] 0.2352941  0.0  0.0
[2,] 0.0000000  0.5 -0.5
[3,] 0.0000000 -0.5  1.0

## Ejercicio Z:

### Ejercicio Z.1

In [22]:
X_Z <- c(0,1,2,3,4,5,6,7,8,0,2,4,6,8,10,0,3,6,9)
X_2Z <-X_Z **2
Y_Z <- c(22.1,24.5,26.0,26.8,28.2,28.9,30.0,30.4,31.4,21.9,26.1,28.5,30.3,31.5,33.1,22.8,27.3,29.8,31.8)
data_Z <- data.frame(X_Z,X_2Z,Y_Z)
data_Z

X_Z X_2Z Y_Z 
1   0    0  22.1
2   1    1  24.5
3   2    4  26.0
4   3    9  26.8
5   4   16  28.2
6   5   25  28.9
7   6   36  30.0
8   7   49  30.4
9   8   64  31.4
10  0    0  21.9
11  2    4  26.1
12  4   16  28.5
13  6   36  30.3
14  8   64  31.5
15 10  100  33.1
16  0    0  22.8
17  3    9  27.3
18  6   36  29.8
19  9   81  31.8

In [31]:
regresion_Z.1 <- lm(Y_Z ~ X_Z+X_2Z, data_Z)
summary(regresion_Z.1)


Call:
lm(formula = Y_Z ~ X_Z + X_2Z, data = data_Z)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.66123 -0.28558 -0.05606  0.34252  0.65440 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 22.56123    0.19843 113.698  < 2e-16 ***
X_Z          1.66802    0.09895  16.857 1.31e-11 ***
X_2Z        -0.06796    0.01031  -6.591 6.21e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3942 on 16 degrees of freedom
Multiple R-squared:  0.9878,	Adjusted R-squared:  0.9863 
F-statistic: 649.9 on 2 and 16 DF,  p-value: 4.782e-16


In [24]:
anova(regresion_Z.1)

Df Sum Sq     Mean Sq     F value    Pr(>F)      
X_Z        1 195.242888 195.2428882 1256.28852 1.235421e-16
X_2Z       1   6.751565   6.7515651   43.44288 6.206331e-06
Residuals 16   2.486599   0.1554125         NA           NA

In [30]:
aov(regresion_Z.1)
summary(aov(regresion_Z.1))

Call:
   aov(formula = regresion_Z.1)

Terms:
                      X_Z      X_2Z Residuals
Sum of Squares  195.24289   6.75157   2.48660
Deg. of Freedom         1         1        16

Residual standard error: 0.3942239
Estimated effects may be unbalanced

            Df Sum Sq Mean Sq F value   Pr(>F)    
X_Z          1 195.24  195.24 1256.29  < 2e-16 ***
X_2Z         1   6.75    6.75   43.44 6.21e-06 ***
Residuals   16   2.49    0.16                     
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

### Ejercicio z.2

In [27]:
qf(0.95, 16, 2)

[1] 19.43329

Teniendose que el valor F empirico es 1299.73 mientras que el valor teorico es 19.4333, con lo que se rechaza H0 y entonces hay algún parametro relevante, de hecho todos lo son.